app/src/test/java/org/kinecosystem/kinit/daggerCore/TestCoreCompImpl.kt

```diff
@@ -0,0 +1,28 @@
+package org.kinecosystem.kinit.daggerCore
+
+import android.content.Context
+import org.kinecosystem.kinit.dagger.ContextModule
+import org.kinecosystem.kinit.dagger.DataStoreProviderModule
+import org.kinecosystem.kinit.mocks.MockDataStoreProviderProvider
+import org.kinecosystem.kinit.repository.DataStoreProvider
+import org.mockito.Mockito.mock
+
+class TestCoreComponentProvider {
```

Comment:
You don't really need a provider that holds an instance of what you create
You only need a method that creates it that can be used from a different number of places.
I would change this to:

class TestCoreComponent {
    fun coreComponent():TestCoreComponent {
         // build the component and return it... 
    }
}

and if in the future we need to create different combination - we could add a method here ?